### PR TITLE
Insertion

### PR DESCRIPTION
**Pull request by rawalparas**
I created an APIs to insert the book author and language data into the collections & to get all the inserted data.

### CODE REVIEW

File: src/model/booksModel.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      ref: 'languages', // Reference to the Language model
+    },
+  });
+  
+const model = mongoose.model("Book", bookSchema);
+module.exports = model;
+
+
+
+function validateName(value) {
```
Comment (athanbhanit) [inline src/model/booksModel.js:29]: Can we reuse this method defined in each model ?

File: src/model/booksModel.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    return regex.name.test(value);
+}
+
+function validateMessageName(props) {
```
Comment (athanbhanit) [inline src/model/booksModel.js:33]: Can we reuse this method defined in each model ?

File: src/API/controller/create/createController.js [hunk 3]
```diff
@@ -1,16 +1,35 @@
...
+const Language = require("../../../model/languageModel.js");
 
 module.exports = {
-    create: async (req, res) => {
-        try {
-            const createBook = await books.create(req.body);
+  insertBooks : async(req , res) => {
+    try{
+      // finding the language name in the language collection.
+      let languageId = await Language.findOne({"name" : req.body.language});
```
Comment (athanbhanit) [inline src/API/controller/create/createController.js:10]: Why Language model name starts with caps but other are not?

File: src/API/controller/create/createController.js [hunk 4]
```diff
@@ -1,16 +1,35 @@
...
 
-            console.log("Inserted data: ", createBook);
-            return res.status(200).send(messages.SUCCESSSFULLY_CREATED);
-        } catch (e) {
-            console.log("Unexpected Error: " + e.message);
-            return res.status(500).send({ message: e.message });
-        }
+      if(!languageId){
+        // inserting the language in the language collection.
+        languageId = await Language.create({"name" : req.body.language});
```
Comment (athanbhanit) [inline src/API/controller/create/createController.js:14]: Why "name" key is defined multiple time? Could it be one?

File: src/API/controller/create/createController.js [hunk 5]
```diff
@@ -1,16 +1,35 @@
...
+      // finding the authorName in the author collection.
+      let authorId = await author.findOne({"name" : req.body.author});
+
+      if(!authorId){
+        authorId = await author.create({"name" : req.body.author});
+      }
+
+      let bookData
+
+      if(languageId._id && authorId._id){
```
Comment (athanbhanit) [inline src/API/controller/create/createController.js:26]: What happens if languageId and authorId is not found?

File: src/API/controller/create/createController.js (hunk 3 above)
Comment (rawalparas) [reply src/API/controller/create/createController.js]: Now It starts with small letter.

File: src/model/booksModel.js (hunk 2 above)
Comment (rawalparas) [reply src/model/booksModel.js]: Yes we can use it. I put this method into the helper folder where i can passing the regex pattern and the values it has to check.

File: src/API/controller/create/createController.js (hunk 5 above)
Comment (rawalparas) [reply src/API/controller/create/createController.js]: It returns a bad request

File: src/API/controller/create/createController.js (hunk 4 above)
Comment (rawalparas) [reply src/API/controller/create/createController.js]: if we have to search in multiple collections then we have to define this

File: src/model/booksModel.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
+const mongoose = require('mongoose');
```
Comment (athanbhanit) [inline src/model/booksModel.js:1]: why file name named as plural?

File: src/helper/validateModelMethod.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+const message = require('../helper/messages.js');
```
Comment (athanbhanit) [inline src/helper/validateModelMethod.js:1]: Make file name as helpermethod

File: src/API/controller/create/createController.js [hunk 8]
```diff
@@ -1,16 +1,42 @@
-const books = require('../../../model/bookModel.js');
-const messages = require('../../../helper/messages.js');
+const author = require("../../../model/authorModel.js");
+const messages = require("../../../helper/messages.js");
+const book = require("../../../model/booksModel.js");
+const language = require("../../../model/languageModel.js");
+const { error } = require("winston");
```
Comment (athanbhanit) [inline src/API/controller/create/createController.js:5]: What is use of winston ?

**Pull request merged**